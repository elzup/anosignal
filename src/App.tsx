import { Box } from 'ink'
import * as React from 'react'
import CodeBox from './components/CodeBox'
import FullWidthMinuteTimer from './components/FullWidthMinuteTimer'
import MapBox from './components/MapBox'
import RandomBox from './components/RandomBox'
import Time from './components/Time'
import Version from './components/Version'
import Status from './components/Status'

const App = () => {
  return (
    <Box flexDirection="column">
      <FullWidthMinuteTimer />
      <Status />
      <Box>
        <Box flexGrow={1}>
          <Time />
        </Box>
        <RandomBox />
      </Box>
      <Box>
        <MapBox />
        <CodeBox />
      </Box>
      <Version />
    </Box>
  )
}

export default App
