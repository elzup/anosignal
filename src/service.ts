import fetch from 'node-fetch'
import * as scrapeIt from 'scrape-it'

type QuestionsResponse = {
  items: {
    tags: string[]
    owner: {}
    is_answered: boolean
    view_count: number
    accepted_answer_id: number
    answer_count: number
    question_id: number
    link: string
    title: string
  }[]
}

const qurl =
  'https://api.stackexchange.com/2.2/questions?order=desc&sort=activity&site=stackoverflow'
export async function getRandomCodes() {
  const res = await fetch(qurl)
  if (!res.ok) return []
  const body: QuestionsResponse = await res.json()

  const codesPromise = body.items.map(async ({ link }) => {
    const { data } = await scrapeIt<{ codetext: string }>(link, {
      codetext: '.postcell pre code',
    })
    return data.codetext.trim()
  })

  return (await Promise.all(codesPromise)).filter(Boolean)
}
