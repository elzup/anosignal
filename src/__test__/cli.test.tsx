import * as React from 'react'
import * as chalk from 'chalk'
import { render } from 'ink-testing-library'
import { advanceBy, advanceTo, clear } from 'jest-date-mock'
import App from '../App'
import { readFileSync } from 'fs'

jest.mock('../service')
import { getRandomCodes } from '../service'

// @ts-ignore
getRandomCodes.mockImplementation(() => new Promise(obj))

const obj = readFileSync(__dirname + '/getRandomCodes.json')

test('render', (t) => {
  advanceTo(new Date(2020, 10, 10, 0, 0, 0))
  const { lastFrame } = render(<App />)

  expect(lastFrame()).toMatchSnapshot('first')
  clear()
})
