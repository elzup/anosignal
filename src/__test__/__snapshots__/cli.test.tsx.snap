// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render: first 1`] = `
"Раѓ
РћїРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћљ
Рћѓrs: 0                                                                                             Рћѓ
Рћѓex: 0                                                                                             Рћѓ
Рћѓhp: 0/0                                                                                           Рћѓ
РћћРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћђРћў
РћїРћљ 1604934000                                                                                  [90m09[39m  [90mРќЈ[39m
РћѓРћѓ c0c0c                                                                                       [90maz[39m  [90mРќЈ[39m
РћѓРћѓ necnmxwazwvzxrou                                                                            [90mAB[39m  [90mРќЈ[39m
РћѓРћѓ OCBOD98A08609QP4                                                                            [90m0z[39m  [90mРќЈ[39m
РћѓРћѓ 1011111101010010101100101110000                                                             [90m0Z[39m  [90mРќЈ[39m
РћѓРћѓ 1133222111211300                                                                            [90ma.[39m  [90mРќЈ[39m
РћѓРћѓ 13752254560                                                                                 [90m0x[39m  [90mРќЈ[39m
РћѓРћѓ 5fa95970                                                                                    [90m01[39m  [90mРќЈ[39m
РћѓРћѓ qjjb00                                                                                      [90ms*[39m  [90mРќЈ[39m
РћѓРћѓ 20380119031408:542549648                                                                    [90m-+[39m  [90mРќЈ[39m
РћћРћў                                                                                             [90m++[39m  [90mРќЈ[39m
 0
 1
 2
 3
 4
 5
 6
0.000"
`;
